gateway/src/apicast/policy/keycloak_role_check/keycloak_role_check.lua

```diff
@@ -152,38 +159,54 @@ local function match_client_roles(scope, context)
   return true
 end
 
-local function scope_check(scopes, context)
-  local uri = ngx.var.uri
-
-  if not context.jwt then
-    return false
-  end
-
-  for _, scope in ipairs(scopes) do
+local function validate_scope_access(scope, context, uri, request_method)
+  for _, method  in ipairs(scope.methods) do
+    -- make a matched method just in case that `ANY` method is defined and
+    -- the mapping rules does not match, the matched_method need to be
+    -- cleared in the next interaction to trigger with the correct method.
+    local matched_method = request_method
```

Comment:
It's a bit unfortunate that we need to do this because `MappingRule` does not provide a way to match no matter what the http method is.

If we added that, maybe we could simplify this part of the code. What do you think?